smolyay/surrogate.py

```diff
@@ -284,7 +303,11 @@ def __call__(self, x):
             else:
                 term = basis(x_scaled)
             value += coeff*term
-        return numpy.squeeze(value)
+        if output_scalar:
+            value = value.item()
+        else:
+            value = numpy.reshape(value,input_shape)
```

Comment:
Can you apply similar logic to what I had above here? For example, I think you can only get a scalar if your input shape was 0d (for a 1d problem) or 1d (for a >= 2d problem). Otherwise, you should discard the last dimension of the input?
```suggestion
        if (self.dimension == 1 and input_shape.ndim == 0) or (self.dimension > 1 and input_shape.ndim == 1):
            value = value.item()
        else:
            value = numpy.reshape(value, input_shape[:-1])
```